src/main/scala/latis/input/HapiJsonAdapter.scala

```diff
@@ -0,0 +1,45 @@
+package latis.input
+
+import latis.model.DataType
+
+/**
+ * Adapter for HAPI JSON datasets.
+ */
+class HapiJsonAdapter(model: DataType, config: TextAdapter.Config = new TextAdapter.Config("dataMarker" -> "\"data\":\\["))
```

Comment:
I’m altering the data marker in the config here so that we can instantiate a `HapiJsonAdapter` with just a `model` and the defaults should work in all cases.

Reply:
Correct me if I'm wrong, but if I remove the `config` from this constructor, my `HapiJsonAdapter` will use the default `TextAdapter` config. Right? If that's so, I don't think that's the right move because the default data marker in `TextAdapter`'s config is not `"data":[`.

Reply:
Could you hardcode that Config as an arg to `TextAdapter` after the "extends"

Reply:
@dlindhol A few things came up with that refactor:
1. This works: `class HapiJsonAdapter(model: DataType)
  extends TextAdapter(model, new TextAdapter.Config("dataMarker" -> "\"data\":\\[")) {`

2. However, doing that disallowed me from accessing the config because now there's no variable in scope for it, so I had to change `splitAtDelim` from `str.trim.split(config.delimiter).toVector` to `str.trim.split(",").toVector` (hardcoding the comma).

3. Now I'm a little confused about the companion object that extends `AdapterFactory`. When you said below "See my comment about the constructor" I took that as "remove `config` from the `apply` constructors too." But when I do, (1) they become identical so I'd remove one, but (2) that doesn't compile because `AdapterFactory`'s `apply` needs both a `model` and a `config`. 
    - So do I remove the companion object altogether? 
    - That doesn't seem like the right thing to do. Do I keep a `config` in the constructor but just not pass it to `new HapiJsonAdapter(model)`? Should `AdapterFactory` be changed to not require a `config` in its `apply`?

Reply:
We could add a `val` to the `config` arg in the `TextAdapter` constructor to make it visible. That should address #2.

Note that `AdapterConfig` is not the same as `TextAdapter.Config`. This is due to some factory construction gymnastics that could be reviewed. The apply with the former is used by the factory (and thus very generic). The apply with the latter is just a "smart" constructor version of the primary constructor. That's the one that should be able to live without the config arg.

Reply:
Done.